dask_jobqueue/tests/test_jobqueue_core.py

```diff
@@ -173,3 +173,31 @@ def test_jobqueue_cluster_call(tmpdir):
     match = re.compile(match, re.DOTALL)
     with pytest.raises(RuntimeError, match=match):
         cluster._call([sys.executable, path_with_error.strpath])
+
+
+@pytest.mark.parametrize(
+    "Cluster",
+    [PBSCluster, MoabCluster, SLURMCluster, SGECluster, LSFCluster, OARCluster],
+)
+def test_cluster_has_cores_and_memory(Cluster):
+    error_message_string = (
+        "You must specify how many cores to use per job like ``cores=8``"
+        + "as well as how much memory to use per job like ``memory='24 GB'``"
+    )
+    with pytest.raises(ValueError, match=error_message_string):
```

Comment:
I would suggest using regexp as I hinted in my previous message. You don't want to check the exact full message but only parts of it. In case someone does a minor wording change of the error message, you'd like the test to still pass (in an ideal world).